src/Aquifer.API/Endpoints/Bibles/Texts/Get/Endpoint.cs

```diff
@@ -51,8 +53,60 @@ public override async Task HandleAsync(Request req, CancellationToken ct)
                         }).ToList()
                 })
                 .ToListAsync(ct);
+            
+            await MapVersificationDifferencesToResponseVersesAsync(req, response, ct);
         }
 
         await (response is null ? SendNotFoundAsync(ct) : SendOkAsync(response, ct));
     }
+
+    private static VerseReference FormatBaseBookChapterVerseMapping(int mappedVerseId)
+    {
+        var (targetBookId, targetChapter, targetVerse) = BibleUtilities.TranslateVerseId(mappedVerseId);
+        var targetBookName = BibleBookCodeUtilities.FullNameFromId(targetBookId);
+        
+        return new VerseReference
+        {
+            BookName = targetBookName,
+            ChapterNumber = targetChapter,
+            VerseNumber = targetVerse
+        };
+    }
+
+    private async Task MapVersificationDifferencesToResponseVersesAsync(Request req, Response response, CancellationToken ct)
+    {
+        if (!response.Chapters.Any())
+        {
+            return;
+        }
+
+        var bookId = BibleBookCodeUtilities.IdFromCode(response.BookCode);
+        var minChapter = response.Chapters.First();  
+        var minVerseId = BibleUtilities.GetVerseId(bookId, minChapter.Number, minChapter.Verses.First().Number);  
+        var maxChapter = response.Chapters.Last();  
+        var maxVerseId = BibleUtilities.GetVerseId(bookId, maxChapter.Number, maxChapter.Verses.Last().Number);
+        
+        var versificationMap = await VersificationUtilities.ConvertVersificationRangeAsync(
+            0, // the "eng" English Bible superset "common" versification
+            minVerseId,
+            maxVerseId,
+            req.BibleId,
+            versificationService,
+            ct);
+        
+        var baseBibleVerseMappings = versificationMap
+            .Where(mapping => mapping.Value.HasValue && mapping.Value.Value != mapping.Key)
+            .ToDictionary(
+                mapping => mapping.Key,
+                mapping => FormatBaseBookChapterVerseMapping(mapping.Value!.Value));
+        
+        foreach (var chapter in response.Chapters)
+        {
+            foreach (var verse in chapter.Verses)
+            {
+                var verseId = BibleUtilities.GetVerseId(bookId, chapter.Number, verse.Number);
+                verse.EnglishBaseVerseMapping = baseBibleVerseMappings.GetValueOrDefault(verseId)!;
```

Comment:
Why is the `!` appended?  `EnglishBaseVerseMapping` allows `null`.
```suggestion
                verse.EnglishBaseVerseMapping = baseBibleVerseMappings.GetValueOrDefault(verseId);
```

Reply:
the `!` was a holdover from the LINQ implementation. It was crying about it but now it is not needed.